Google.PowerShell/Sql/GcSqlInstanceCmdlets.cs

```diff
@@ -513,18 +535,48 @@ private class ParameterSetNames
         [Parameter(Mandatory = false, Position = 4, ParameterSetName = ParameterSetNames.Csv)]
         public string[] Column { get; set; }
 
+        /// <summary>
+        /// <para type="description">
+        ///  If true, uploads the object at ImportFilePath to a new Google Cloud Storage bucket, then imports it
+        ///  into the instance. Otherwise, imports the file at the Google Cloud Storage location
+        ///  specified by ImportFilePath
+        /// </para>
+        /// </summary>
+        [Parameter]
+        public SwitchParameter UploadLocalFile { get; set; }
+
+        /// <summary>
+        /// MIME attachment for general binary data. (Octets of bits, commonly referred to as bytes.)
+        /// </summary>
+        protected const string OctetStreamMimeType = "application/octet-stream";
+
+        private string bucketName;
```

Comment:
Adding three class properties is semi dubious, since it is unclear what their lifecycle will be. (e.g. who will initialize bucketService, when will it be closed, etc.?)

Consider creating a private class "GcsFileUploader". In that class's constructor it initializes the StorageSerivce object. And it has several methods "CreateBucket(string bucketName)", "UploadLocalFile(string filePath, DefaultObjectAcls acls)", "DeleteObject(string bucketName, string objectPath)", and "DeleteBucket(string bucketName)".

Doing it that way enables that part of the implementation to be designed, built, and tested independently.

It also reduces the complexity of the parent cmdlet by removing various fields.